mitosheet/mitosheet/code_chunks/step_performers/dataframe_steps/dataframe_delete_code_chunk.py

```diff
@@ -3,22 +3,93 @@
+                    second_sheet_indexes[index] = second_sheet_index + 1
+
+        sheet_indexes = first_sheet_indexes + second_sheet_indexes
+        old_dataframe_names = self.get_param('old_dataframe_names') + other_code_chunk.get_param('old_dataframe_names')
+
+        return DataframeDeleteCodeChunk(
+            self.prev_state,
+            other_code_chunk.post_state,
+            {
+                'sheet_indexes': sheet_indexes,
+                'old_dataframe_names': old_dataframe_names,
+            },
+            {}
+        )
+
+
+    def combine_right(self, other_code_chunk: "CodeChunk") -> Optional["CodeChunk"]:
+        if isinstance(other_code_chunk, DataframeDeleteCodeChunk):
+            return self._combine_right_dataframe_delete(other_code_chunk)
+        
+        return None
+
+    def combine_left(self, other_code_chunk: "CodeChunk") -> Optional["CodeChunk"]:
+        deleted_sheet_indexes = self.get_param('sheet_indexes')
+        created_sheet_indexes = other_code_chunk.get_created_sheet_indexes()
+        print(deleted_sheet_indexes, created_sheet_indexes)
+        if created_sheet_indexes and set(deleted_sheet_indexes) == set(created_sheet_indexes):
+            # If all we did was creat the dfs we deleted, we can just return a no op
+            return NoOpCodeChunk(other_code_chunk.prev_state, self.post_state, {}, {})
+            
+        elif created_sheet_indexes and set(deleted_sheet_indexes).issuperset(set(created_sheet_indexes)):
+            # If the set we are deleting is a superset of the the dataframes that we created
+            # in this other step, then we must create a new code chunk that doesn't have these
+            # deletes for the created sheets (as we're just not including these created steps)
+            new_deleted_sheet_indexes = copy(self.get_param('sheet_indexes'))
+            new_old_dataframe_names = copy(self.get_param('old_dataframe_names'))
+
+            for created_sheet_index in created_sheet_indexes:
+                # Find the index of the sheet index we need to delete, and remove that as well
+                # as the old dataframe name, which is at the same index
+                index_in_params_array = new_deleted_sheet_indexes.index(created_sheet_index)
+                new_deleted_sheet_indexes.pop(index_in_params_array)
+                new_old_dataframe_names.pop(index_in_params_array)
+            
+            return DataframeDeleteCodeChunk(
+                other_code_chunk.prev_state,
+                self.post_state,
+                {
+                    'sheet_indexes': new_deleted_sheet_indexes,
+                    'old_dataframe_names': new_old_dataframe_names,
+                },
+                {}
+            )
+
+        sheet_indexes_edited = other_code_chunk.get_edited_sheet_indexes()
+        if sheet_indexes_edited and set(deleted_sheet_indexes).issuperset(set(sheet_indexes_edited)):
+            return self
```

Comment:
Okay, so instead of returning `False` for the rename column dtype, maybe here we could just case on the type, like: 

```
if sheet_indexes_edited and set(deleted_sheet_indexes).issuperset(set(sheet_indexes_edited)) and not
other_code_chunk.get_display_name() == 'Renamed columns`:
```